index.js

```diff
@@ -1,18 +1,138 @@
-const readline = require('readline');
+const { exit } = require("process");
+const readline = require("readline");
 const rl = readline.createInterface(process.stdin, process.stdout);
 
+// prompt user with question func
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     rl.question(questionText, resolve);
   });
 }
 
+//Get a random number in a range
+function randomNumGen(min, max) {
+  return Math.floor(Math.random() * (max - min) + min);
+}
+
 start();
 
+// Makes array 0 to max
+function createArray(max) {
+  let gameArray = [];
+  for (let i = 1; i <= parseInt(max); i++) {
+    gameArray.push(i);
+  }
+
+  return gameArray;
+}
+
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
-  process.exit();
+  // Geting user input for gameArray range and asking them which of the two games they'd like to play
+  let whatGame = await ask(
+    "Would you, the human (H) like to guess my secret number? Or should I the computer (C) guess yours?"
+  );
+
+  let upperBound = await ask("How many numbers would you like to use?");
```

Comment:
this is another example of where a comment would be helpful. what is upperBound? why do we need it?